lib/connect/index.js

```diff
@@ -105,8 +105,13 @@ function connect (brokerUrl, opts) {
   }
 
   if (!protocols[opts.protocol]) {
-    opts.protocol = protocolList.filter(function (key) {
-      return typeof protocols[key] === 'function'
+    const isSecure = (['mqtts', 'wss'].indexOf(opts.protocol) !== -1)
+    opts.protocol = protocolList.filter(function (key, index) {
+      if (isSecure && index % 2 === 0) {
```

Comment:
I don't like this dependency on order in the protocol array. If the order changes in the array or other protocols are added this might break. Maybe have two arrays, one for secure protocols and one for insecure protocols, which in turn protocolList is built from.